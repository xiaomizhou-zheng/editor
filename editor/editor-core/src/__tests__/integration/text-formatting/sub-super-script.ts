import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';
import {
  fullpage,
  editable,
  getDocFromElement,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import superScriptAdf from './__fixtures__/super-script.adf.json';

BrowserTestCase(
  'Text formatting: It can load a document with super script on first line',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: superScriptAdf,
    });

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
