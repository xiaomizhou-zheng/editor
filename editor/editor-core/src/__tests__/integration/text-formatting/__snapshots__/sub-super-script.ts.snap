// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text formatting: It can load a document with super script on first line 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "type": "sup",
              },
              "type": "subsup",
            },
          ],
          "text": "Will not load",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
